tracee-rules/engine/engine.go

```diff
@@ -108,17 +124,81 @@ func (engine *Engine) consumeSources(done <-chan bool) {
 						}
 					}
 				}
+				engine.signaturesMutex.RUnlock()
 				engine.inputs.Tracee = nil
 			} else if event != nil {
+				engine.signaturesMutex.RLock()
 				for _, s := range engine.signaturesIndex[types.SignatureEventSelector{Source: "tracee", Name: event.(tracee.Event).EventName}] {
 					engine.signatures[s] <- event
 				}
 				for _, s := range engine.signaturesIndex[types.SignatureEventSelector{Source: "tracee", Name: "*"}] {
 					engine.signatures[s] <- event
 				}
+				engine.signaturesMutex.RUnlock()
 			}
 		case <-done:
 			return
 		}
 	}
 }
+
+//LoadSignature will store in Engine data structures the given signature and activate its handling business logics.
+// It will return the signature ID as well as error.
+func (engine *Engine) LoadSignature(signature types.Signature) (string, error) {
```

Comment:
I see that you went with just load/unload API which also does the startsig/stopsig we discussed. That's fine with me I think just wanted to clarify you made this is intentional

Reply:
yes, it is.